### ISSUE TITLE
Shareable Configs passed throu cli (-c, --config) does not work

### ISSUE THREAD

**Issue by cojj90** [eslint/eslint#10105]
**Tell us about your environment**
Using sharable config throughout our repos 

* **ESLint Version:**
4.19.0
* **Node Version:**
8.9.4
* **npm Version:**
5.7.1
**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  extends: [
    'standard',
    'standard-jsx',
    'standard-babel',
    'plugin:flowtype/recommended'
  ],
  plugins: [
    'babel',
    'standard',
    'flowtype'
  ],
  rules: {
    // Require braces in arrow function body
    // http://eslint.org/docs/rules/arrow-body-style
    'arrow-body-style': 0,

    // Require parens in arrow function
    // http://eslint.org/docs/rules/arrow-parens
    'arrow-parens': [2, 'always'],

    // Enforce consistent spacing inside braces
    // http://eslint.org/docs/rules/object-curly-spacing
    'object-curly-spacing': [2, 'always']
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash
.../node_modules/eslint/bin/eslint.js --config <SHARED_CONFIG> --fix src
```

**What did you expect to happen?**
use the sharable config

**What actually happened? Please include the actual, raw output from ESLint.**
it does not make use of shared config passed from the CLI. 

from debug log
```
  eslint:cli Running on files +0ms
  eslint:config Using command line config <SHARED_CONFIG> +0ms
  eslint:config-file Attempting to resolve eslint-config-<SHARED_CONFIG> +0ms
  eslint:config-file Loading JS config file: .../node_modules/eslint-config-<SHARED_CONFIG>/index.js +1ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in ... +0ms
  eslint:ignored-paths Could not find ignore file in cwd +0ms
  eslint:cli-engine Linting complete in: 5ms +0ms
  eslint:cli Fix mode enabled - applying fixes +151ms
```

**Comment by platinumazure** [comment]
Hi @cojj90, thanks for the issue.

I'm not sure I understand. I do see the config file being loaded. I don't see any debug statements indicating that any file is being linted. Have you provided the whole debug log?

Are you sure you're running this from the right working directory?

**Comment by cojj90** [comment]
Hi @platinumazure - sorry for wording that poorly.

According to the debug log - the config file gets loaded but it clearly doesn't get applied.
I can tell that because `parser: 'babel-eslint'` isn't applied (which is in my shared config).

I can fix this issue by creating a .eslintrc on my root project and inserting ` parser: 'babel-eslint'` but this will defeat the purpose of shareable config.

**Comment by galvarez421** [comment]
I am experiencing this same issue.

node_modules/eslint-config-myshareableconfig/index.js:

```
module.exports = {
  env: {
    node: true,
  },
  extends: ['eslint:recommended', 'google'],
  parserOptions: {
    ecmaVersion: 6,
    ecmaFeatures: {
      impliedStrict: true,
    },
  },
  rules: {
    'linebreak-style': 0,
    'max-len': [1, {
      code: 80,
      tabWidth: 2,
      ignoreUrls: true,
      ignorePattern: '^goog\.(module|require)',
    }],
    'spaced-comment': [1, 'always'],
  },
};
```

Running the following command throws `Parsing error: The keyword 'const' is reserved` error.

`eslint --no-eslintrc --color --cache --config myshareableconfig webpack-common.config.js`

As @cojj90 experienced, the shareable ID seems to be recognized (changing `myshareableconfig` to `madeup` throws error that file is not found) but not applied (with ecmaVersion set to 6, const should be allowed).

**Comment by galvarez421** [comment]
I fixed this issue in my case by changing the `--config` argument to be `node_modules/eslint-config-myshareableconfig/index.js` instead of the shareable config shorthand ID `myshareableconfig`. I believe there is still an issue, however, because the docs state that the shorthand ID is allowed in the CLI (https://eslint.org/docs/user-guide/command-line-interface#-c---config).

I should mention that the ESLint command is being run by running `yarn run` for a package.json script (both when it didn't work and now that it does), in case that has relevance.

Here is the new debug information (displayed because of an _expected_ lint warning rather than the _unexpected_ `const` error):

```
  eslint:cli Running on files +0ms
  eslint:config Using command line config node_modules/eslint-config-myshareableconfig/index.js +0ms
  eslint:config-file Loading JS config file: C:\myproject\node_modules\eslint-config-myshareableconfig\index.js +0ms
  eslint:config-file Loading google +291ms
  eslint:config-file Attempting to resolve eslint-config-google +1ms
  eslint:config-file Loading JS config file: C:\myproject\node_modules\eslint-config-google\index.js +3ms
  eslint:config-file Loading C:\myproject\node_modules\eslint\conf\eslint-recommended.js +85ms
  eslint:config-file Loading JS config file: C:\myproject\node_modules\eslint\conf\eslint-recommended.js +0ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in C:\myproject +0ms
  eslint:ignored-paths Could not find ignore file in cwd +1ms
  eslint:config Constructing config file hierarchy for C:\myproject +402ms
  eslint:config Not using .eslintrc or package.json files +1ms
  eslint:config Using command line config file +0ms
  eslint:config-ops Using config from partial cache +0ms
  eslint:config-ops Apply environment settings to config +1ms
  eslint:config-ops Creating config for environment node +1ms
  eslint:cli-engine Processing C:\myproject\webpack-common.config.js +0ms
  eslint:cli-engine Linting C:\myproject\webpack-common.config.js +2ms
  eslint:config Constructing config file hierarchy for C:\myproject +7ms
  eslint:config Not using .eslintrc or package.json files +1ms
  eslint:config Using command line config file +0ms
  eslint:config Using config from cache +1ms
  eslint:linter Linting code for C:\myproject\webpack-common.config.js (pass 1) +0ms
  eslint:linter Generating fixed text for C:\myproject\webpack-common.config.js (pass 1) +78ms
  eslint:text-fixer Applying fixes +0ms
  eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:config Constructing config file hierarchy for C:\myproject +82ms
  eslint:config Not using .eslintrc or package.json files +0ms
  eslint:config Using command line config file +1ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Linting complete in: 106ms +89ms
```

One thing that I find interesting and confusing is that both commands result in the same shareable config file being loaded, but the ` eslint:config-ops Creating config for environment node` message is displayed only when I change the `--config` argument as described above, i.e., when the specified shareable config is used as expected.
